timdex_dataset_api/record.py

```diff
@@ -19,35 +19,51 @@ class DatasetRecord:
     timdex_record_id: str
     source_record: bytes
     transformed_record: bytes
+    source: str
+    run_date: str | datetime.datetime
+    run_type: str
+    run_id: str
+    action: str
 
     # partition columns
-    source: str | None = None
-    run_date: str | datetime.datetime | None = None
-    run_type: str | None = None
-    run_id: str | None = None
-    action: str | None = None
+    year: str | None = None
+    month: str | None = None
+    day: str | None = None
+
+    def __post_init__(self) -> None:
```

Comment:
See comment above about potentially making `year`, `month`, and `day` dynamic properties, which would have bearing on this method.

Reply:
All of our shared learnings from our many discussions led to the updates in this commit: https://github.com/MITLibraries/timdex-dataset-api/pull/15/commits/5e532d365baf9b6f6db68d68dd09d20dabfece00. Please take a look at the commit message, which pulls from our discussion and provides more context for the changes introduced!